move/squid/tests/squid/deepbook_v3_tests.move

```diff
@@ -0,0 +1,55 @@
+#[test_only]
+module squid::deepbook_v3_tests {
+use std::type_name::{Self, TypeName};
+    use sui::sui::SUI;
+    use sui::clock::{Self, Clock};
+    use sui::object::id_to_address;
+    use token::deep::DEEP;
+    use deepbook::pool::Pool;
+    use squid::deepbook_v3::{Self, DeepbookV3SwapData};
+    use squid::swap_info;
+    use sui::test_scenario::{Scenario, begin, end, return_shared};
+    use sui::test_utils::{destroy, assert_eq};
+
+    use deepbook::pool_tests;
+
+    public struct USDC has store {}
+
+    const OWNER: address = @0x1;
+
+    #[test]
+    fun test_serialize() {
+        let mut test = begin(OWNER);
+        let pool_id = pool_tests::setup_everything<
+            SUI,
+            USDC,
+            SUI,
+            DEEP,
+        >(&mut test);
+
+        test.next_tx(OWNER);
+        let swap_data = deepbook_v3::new_swap_data(
+            1,
+            id_to_address(&pool_id),
+            true,
+            100,
+            type_name::get<SUI>().into_string(),
+            type_name::get<USDC>().into_string(),
+            1,
+            true,
+        );
```

Comment:
How do I convert this into `data` to create a `SwapInfo` object?

Reply:
`bcs::serialize(swap_data)` should do the trick.